lib/shopify-cli/theme/development_theme.rb

```diff
@@ -58,7 +69,10 @@ def generate_theme_name
         hostname = Socket.gethostname.split(".").shift
         hash = SecureRandom.hex(3)
 
-        theme_name = "Development (#{hash}-#{hostname})"
+        theme_name = "Development ()"
+        hostname_character_limit = API_NAME_LIMIT - theme_name.length - hash.length - 1
+        identifier = "#{hash}-#{hostname.truncate(hostname_character_limit)}"
+        theme_name = theme_name.gsub("()", "(#{identifier})")
```

Comment:
That won't work if `theme_name` ends up containing another `()`. Either from hostname or future code modifications.

```suggestion
        theme_name = "Development (#{identifier})"
```

Reply:
That's a good point. I committed your suggestions @macournoyer 🙏🏼